exporter/clickhouseexporter/factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clickhouseexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickhouseexporter"
+
+import (
+	"context"
+	"fmt"
+
+	_ "github.com/ClickHouse/clickhouse-go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "clickhouse"
+)
+
+// NewFactory creates a factory for Elastic exporter.
+func NewFactory() component.ExporterFactory {
+	return exporterhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporterhelper.WithLogs(createLogsExporter),
+	)
+}
+
+func createDefaultConfig() config.Exporter {
+	return &Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		Address:          "tcp://127.0.0.1:9000",
+		Database:         "default",
+	}
+}
+
+// createLogsExporter creates a new exporter for logs.
+//
+// Logs are directly indexed into clickhouse.
+func createLogsExporter(
+	ctx context.Context,
+	set component.ExporterCreateSettings,
+	cfg config.Exporter,
+) (component.LogsExporter, error) {
+	exporter, err := newExporter(set.Logger, cfg.(*Config))
+	if err != nil {
+		return nil, fmt.Errorf("cannot configure clickhouse logs exporter: %w", err)
+	}
+
+	return exporterhelper.NewLogsExporter(
+		cfg,
+		set,
+		exporter.pushLogsData,
+		exporterhelper.WithShutdown(exporter.Shutdown),
```

Comment:
There is additional helpers that you can adopt here as well, I would suggest looking at the following ones:

- `exporterhelper.WithTimeout`
- `exporterhelper.WithRetry`

This would help make the exporter more durable / reliable for those whom adopt it.